### PR TITLE
Promote important architecture description that answers a lot of questions we get

### PR DESCRIPTION
**Pull request by bf4**
I think that this docs was generally being ignored and could answer a lot of questions we've been getting. So, I promoted it to the README for more visibility

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -156,7 +156,141 @@ serializer = SomeSerializer.new(resource, serializer_options)
...
+### ActiveModel::Serializer
+
+An **`ActiveModel::Serializer`** wraps a [serializable resource](https://github.com/rails/rails/blob/4-2-stable/activemodel/lib/active_model/serialization.rb)
+and exposes an `attributes` method, among a few others.
+It allows you to specify which attributes and associations should be represented in the serializatation of the resource.
+It requires an adapter to transform its attributes into a JSON document; it cannot be serialized itself.
+It may be useful to think of it as a
+[presenter](http://blog.steveklabnik.com/posts/2011-09-09-better-ruby-presenters).
+
+#### ActiveModel::CollectionSerializer
```
Comment (bf4) [inline README.md]: So we can link to these.

File: README.md [hunk 2]
```diff
@@ -156,7 +156,141 @@ serializer = SomeSerializer.new(resource, serializer_options)
...
+API](http://jsonapi.org/) document.
+
+### ActiveModelSerializers::SerializableResource
+
+The **`ActiveModelSerializers::SerializableResource`** acts to coordinate the serializer(s) and adapter
+to an object that responds to `to_json`, and `as_json`.  It is used in the controller to
+encapsulate the serialization resource when rendered. However, it can also be used on its own
+to serialize a resource outside of a controller, as well.
+
+### Primitive handling
```
Comment (bf4) [inline README.md]: could use some attention to rewrite, but I didn't want to expand the scope here

File: docs/general/rendering.md [hunk 3]
```diff
@@ -48,26 +48,11 @@ render json: @posts, serializer: CollectionSerializer, each_serializer: PostPrev
 
 ## Serializing non-ActiveRecord objects
 
-All serializable resources must pass the
-[ActiveModel::Serializer::Lint::Tests](../../lib/active_model/serializer/lint.rb#L17).
-
-See the ActiveModelSerializers::Model for a base class that implements the full
-API for a plain-old Ruby object (PORO).
+See [README](../../../README.md)
```
Comment (bf4) [inline docs/general/rendering.md]: `See [README](../../README.md#activemodelserializersserializableresource)`

Comment (bf4) [inline docs/general/rendering.md]: And [README](../../README.md#what-does-a-serializable-resource-look-like)`

File: README.md (hunk 2 above)
Comment (NullVoxPopuli) [inline README.md]: at least this is something easily point-to-able from people wanting to serialize hashes and such

**Comment by NullVoxPopuli** [comment]
I'm a fan of this information being more easily accessible.

**Pull request merged**